tezos_interop/tezos_interop.re

```diff
@@ -693,43 +693,100 @@ module Consensus = {
         // TODO: proper type for amounts
         amount: Z.t,
         destination: Address.t,
-      });
+      })
+    | Update_root_hash(BLAKE2B.t);
   type operation = {
     hash: BLAKE2B.t,
     index: int,
     parameters,
   };
 
-  let parse_parameters = (entrypoint, micheline) =>
-    switch (entrypoint, micheline) {
-    | (
-        "deposit",
-        Tezos_micheline.Micheline.Prim(
-          _,
-          Michelson_v1_primitives.D_Pair,
-          [
-            Bytes(_, destination),
+  let parse_parameters = (entrypoint, micheline) => {
+    Michelson_v1_primitives.(
+      switch (entrypoint, micheline) {
+      | (
+          "update_root_hash",
+          Tezos_micheline.Micheline.(
             Prim(
               _,
               D_Pair,
               [
-                Bytes(_, ticketer),
-                Prim(_, D_Pair, [Bytes(_, data), Int(_, amount)], _),
+                Prim(
+                  _,
+                  D_Pair,
+                  [
+                    Prim(
+                      _,
+                      D_Pair,
+                      [Bytes(_, _block_hash), Int(_, _block_height)],
+                      _,
+                    ),
+                    Prim(
+                      _,
+                      D_Pair,
+                      [
+                        Bytes(_, _block_payload_hash),
+                        Int(_, _handles_hash),
+                      ],
+                      _,
+                    ),
+                  ],
+                  _,
+                ),
+                Prim(
+                  _,
+                  D_Pair,
+                  [
+                    Prim(
+                      _,
+                      D_Pair,
+                      [_signatures, Bytes(_, state_root_hash)],
+                      _,
+                    ),
+                    _,
+                  ],
+                  _,
+                ),
               ],
               _,
-            ),
-          ],
-          _,
-        ),
-      ) =>
-      let.some destination =
-        Data_encoding.Binary.of_bytes_opt(Address.encoding, destination);
-      let.some ticketer =
-        Data_encoding.Binary.of_bytes_opt(Address.encoding, ticketer);
-      let ticket = Ticket.{ticketer, data};
-      Some(Deposit({ticket, destination, amount}));
-    | _ => None
-    };
+            )
+          ),
+        ) =>
+        switch (state_root_hash |> Bytes.to_string |> BLAKE2B.of_raw_string) {
+        | Some(state_root_hash) => Some(Update_root_hash(state_root_hash))
```

Comment:
As you already extracted all the data, I would ask you to make a record like it's doing on deposit

Reply:
Not sure I understand - you mean this `Some(Update_root_hash({ state_root_hash } ))` ? What do we gain?

Reply:
`Update_root_hash` is not only `state_root_hash`, I'm talking about all the extracted data, not only state_root_hash, again, look how it's doing on deposits.

Reply:
As the other properties are not relevant to the scope of the PR, which is to merely extract out `state_root_hash` so that in a future PR, a lookup can be made to know if the node is in a sync state or not, can we extract the other properties in a future PR?

Adding new properties will not add more things to test in this PR, thereby increasing its scope.

Reply:
@callistonianembrace feel free to open a follow up issue, the idea is that this variant should be 1 : 1 with the variant encoded in the consensus.mligo.

Reply:
Created a follow up issue here - https://github.com/marigold-dev/sidechain/issues/181